### PR TITLE
fix:resolve profiles in local feed

### LINKED ISSUE #898: Profiles not resolved in local feed [bug]
image

https://damus.io/note1mw5mvs6f24e4fvxfppeyn5ww92e5eum807ma9r7q08c8c0jsh2yqu3xt5d

image

image

### PR DESCRIPTION
**Pull request by gislik**
## Scope

This PR fixes the issue of missing profile information of boosted events in the local feed. 

In the process, I discovered that profiles were also missing from boosted events in the profile view, i.e. when you're navigating to your own or other people's profiles, and I decided to add that issue to the scope.

Closes #898 

## Implementation

The proposed solution tries to achieve the desired outcome with minimal code changes. Especially since this is the first Swift code I write and I'm still understanding how to write idiomatic Swift code. I also tried to mimic the patterns I've seen while familiarising myself with the code base.

The main change is addition of the following code in the function `find_profiles_to_fetch_from_events`.

~~~swift
  // lookup profiles from boosted events
  if let bev = ev.inner_event {
      if profiles.lookup(id: bev.pubkey) == nil {
          pubkeys.insert(bev.pubkey)
      }
  }
~~~        

The two other changes simply initiate the lookup from both the home model and the profile model.

## How to test

**Local feed**
1. Click on the **Home** button at the bottom of the screen to load the local feed
2. Click **Posts & Replies**
3. Scroll down until you see a boosted event
4. Notice that the profile of the booster has been resolved/looked up
image

**Profile view**
1. Click on the **Magnifying glass** at the bottom of the screen to load the global feed
2. Search for `npub1pnqv7krwhm2a26p3tdv9pzwgfveqkrp60um6hxaf63vqxsrlhwwqkddhfc`
3. Scroll down until you see a boosted event
image

I also tested by importing the following public keys, who have many followers and follow many people, to try to break the solution.

`@jack npub1sg6plzptd64u62a878hep2kev88swjh3tw00gjsfl8f237lmu63q0uf63m`
`@jb55 npub1xtscya34g58tk0z605fvr788k263gsu6cy9x0mhnm87echrgufzsevkk5s`

### CODE REVIEW

**Comment by gislik** [comment]
I decided to keep the 3 commits separate to help with the review but I suggest squash merging once the PR is accepted 👻

File: damus/Models/SearchHomeModel.swift [hunk 1]
```diff
@@ -128,11 +128,16 @@ func find_profiles_to_fetch_from_events(profiles: Profiles, events: [NostrEvent]
     var pubkeys = Set<String>()
     
     for ev in events {
-        if profiles.lookup(id: ev.pubkey) != nil {
-            continue
+        // lookup profiles from boosted events
+        if let bev = ev.inner_event {
```
[the remark this answers is not in the document]
Comment (gislik) [reply damus/Models/SearchHomeModel.swift:132]: Sure thing. I didn't know that was possible and I'm still getting a feeling for what idiomatic Swift code looks like. 

I haven't seen many instances of this pattern in the Damus code base but happy to change if you feel like it is easier to read 🙌

**Comment by bryanmontz** [comment]
Great PR, by the way! The documentation is more than the code! I'm not the maintainer here, just a contributor. But I've written Swift for many years. This way is easier to read and a better pattern IMO, but it's all up to @jb55. It's not a big deal when it's only two expressions, but imagine how it would look if it were four.

**Comment by gislik** [comment]
Thanks for the kind words 🙏 As a reviewer my self, I have a pretty good idea how to make the life of the reviewer a little bit easier 😀

Anyways, I hope to be able to contribute more to the ecosystem, so all Swift pointers are greatly appreciated.

**Comment by gislik** [comment]
> This way is easier to read and a better pattern IMO, but it's all up to @jb55. 

Just pushed the changes @bryanmontz proposed in a separate commit. Now @jb55 can  easily choose which version he likes by reverting or not reverting [0384c1a](https://github.com/damus-io/damus/pull/947/commits/0384c1a7ba79f43db79c90503102ee0025ca4b3c) 🫡

**Comment by jb55** [comment]
thank you for making logically distinct commits (except for the last one but I forgive you and fixed it up)

**Comment by jb55** [comment]
cherry picked: 209b23674d0a668304fc9b6a6c1f2255c993ba59...5a1b9661918689e63e4908b8037521b1ec8fbaef

**Comment by gislik** [comment]
> thank you for making logically distinct commits (except for the last one but I forgive you and fixed it up)

I already checked that condition,`bev.inner_event` only exists for boosts.

https://github.com/damus-io/damus/blob/eaec3ae011f75138dde3b2187bbb9ddfce2548ef/damus/Nostr/NostrEvent.swift#L114-L120

**Pull request closed without merging**